sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/implementation/ChangeFeedStartFromInternalDeserializer.java

```diff
@@ -0,0 +1,110 @@
+
+import java.io.IOException;
+import java.time.Instant;
+import java.util.Locale;
+
+public class ChangeFeedStartFromInternalDeserializer extends StdDeserializer<ChangeFeedStartFromInternal> {
+
+    private static final long serialVersionUID = 1L;
+
+    public ChangeFeedStartFromInternalDeserializer() {
+        this(null);
+    }
+
+    public ChangeFeedStartFromInternalDeserializer(final Class<?> vc) {
+        super(vc);
+    }
+
+    @Override
+    public ChangeFeedStartFromInternal deserialize(
+        final JsonParser parser,
+        final DeserializationContext context)
+        throws IOException {
+
+        final JsonNode rootNode = parser.getCodec().readTree(parser);
+        final ObjectMapper mapper = (ObjectMapper)parser.getCodec();
+
+        return deserializeStartFromInternal(rootNode, mapper, parser);
+    }
+
+    public static ChangeFeedStartFromInternal deserializeStartFromInternal(
+        JsonNode rootNode,
+        ObjectMapper mapper,
+        JsonParser parser) throws IOException {
+
+        JsonNode typeNode = rootNode.get(Constants.Properties.CHANGE_FEED_START_FROM_TYPE);
+        if (typeNode == null || !typeNode.isTextual() || Strings.isNullOrWhiteSpace(typeNode.textValue())) {
+            throw JsonMappingException.from(
+                parser,
+                String.format(
+                    "Required property '%s' missing or does not have a valid value.",
+                    Constants.Properties.CHANGE_FEED_START_FROM_TYPE));
+        }
+
+        ChangeFeedStartFromTypes type;
+        try {
+            type = ChangeFeedStartFromTypes.valueOf(typeNode.textValue().toUpperCase(Locale.ROOT));
+        } catch (IllegalArgumentException argException) {
+            throw JsonMappingException.from(
+                parser,
+                String.format(
+                    "Value '%s' for required property '%s' does not have a valid value.",
+                    typeNode.textValue(),
+                    Constants.Properties.CHANGE_FEED_START_FROM_TYPE),
+                argException);
+        }
+
+        switch (type) {
+            case BEGINNING:
+                return new ChangeFeedStartFromBeginningImpl();
+
+            case NOW:
+                return new ChangeFeedStartFromNowImpl();
+
+            case POINT_IN_TIME:
+                JsonNode pitNode =
+                    rootNode.get(Constants.Properties.CHANGE_FEED_START_FROM_POINT_IN_TIME_MS);
+                if (pitNode == null || !pitNode.isLong()) {
+                    throw JsonMappingException.from(
+                        parser,
+                        String.format(
+                            "Required property '%s' missing or does not have a valid value.",
+                            Constants.Properties.CHANGE_FEED_START_FROM_POINT_IN_TIME_MS));
+                }
+                return new ChangeFeedStartFromPointInTimeImpl(Instant.ofEpochMilli(pitNode.longValue()));
+
+            case LEASE:
```

Comment:
How does this type work?

Reply:
Code is in ChangeFeedStartFromETagAndFeedRangeImpl - basically a way to bi-directionally map the lease docuemnts V1 to ChangeFeedStateV1 - so that I can re-use it even for ChangeFeedProcessor.

My main idea was that I would allow customers to split continuations - like allowing to process same physical partition via multiple compute instances on the ChangeFeedState - and would disallow it for this state.

But I admit that I haven't finalized the proposal on how to expsoe the split/merge functionality for continuations (and whether it would be preferrable to have that unified for CF vs. query or separately) So I might be able to get rid of this type at some point - it is in the implementation package - so that is possible.